### PR TITLE
arm: stm32: add STM32L4 chips

### PR DESCRIPTION
**Pull request by Kmart777**

### CODE REVIEW

File: nuttx/arch/arm/include/stm32/stm32l4x1xx_irq.h [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/****************************************************************************************************
+ * arch/arm/include/stm32/stm32l4x1xx_irq.h
+ *
+ *   Copyright (C) 2015 Motorola Mobility, LLC. All rights reserved.
```
Comment (jwylder1) [inline nuttx/arch/arm/include/stm32/stm32l4x1xx_irq.h:4]: 2017

File: nuttx/arch/arm/src/stm32/Kconfig [hunk 2]
```diff
@@ -33,7 +33,7 @@ comment "STM32 Configuration Options"
 
 choice
 	prompt "STM32 Chip Selection"
-	default ARCH_CHIP_STM32F103ZE
```
Comment (jwylder1) [inline nuttx/arch/arm/src/stm32/Kconfig]: why are you deleting this one?

**Comment by Kmart777** [comment]
Did not delete, just replaced default to chip that is used in MDK

comment "STM32 Configuration Options"

choice
               prompt "STM32 Chip Selection"
               default ARCH_CHIP_STM32L476MG
               depends on ARCH_CHIP_STM32

**Comment by Kmart777** [comment]
pending new updates that maintain variable names within Kconfig files so that no .c .h files need to be modified.

**Pull request closed without merging**